src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -2627,6 +2627,19 @@ public static Object callIterator(Object obj, Context cx, Scriptable scope)
         return getIterator.call(cx, scope, iterable, ScriptRuntime.emptyArgs);
     }
 
+    /**
+     * Given an iterator result, return true if and only if there is a "done"
+     * property that's true.
+     */
+    public static boolean isIteratorDone(Context cx, Object result)
+    {
+        if (!(result instanceof Scriptable)) {
+            return false;
+        }
+        final Object prop = getObjectProp((Scriptable)result, "done", cx);
+        return toBoolean(prop);
```

Comment:
Maybe a constant for "done" somewhere

Reply:
I was being contrarian there since it's not like we're going to have to change the value of "done" some day. But I agree that it probably makes sense to constant-ize it!

Reply:
But the string is used at several places with the same semantics.